builtin/logical/consul/path_token.go

```diff
@@ -61,25 +61,54 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 
 	writeOpts := &api.WriteOptions{}
 	writeOpts = writeOpts.WithContext(ctx)
-
-	// Create it
-	token, _, err := c.ACL().Create(&api.ACLEntry{
-		Name:  tokenName,
-		Type:  result.TokenType,
-		Rules: result.Policy,
-	}, writeOpts)
-	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+	var s *logical.Response
+	// Create an ACLEntry for Consul pre 1.4
+	if (result.Policy != "" && result.TokenType == "client") || (result.Policy == "" && result.TokenType == "management") {
+		token, _, err := c.ACL().Create(&api.ACLEntry{
+			Name:  tokenName,
+			Type:  result.TokenType,
+			Rules: result.Policy,
+		}, writeOpts)
+		if err != nil {
+			return logical.ErrorResponse(err.Error()), nil
+		}
+		// Use the helper to create the secret
+		s = b.Secret(SecretTokenType).Response(map[string]interface{}{
+			"token": token,
+		}, map[string]interface{}{
+			"token":   token,
+			"role":    role,
+			"version": "1.3",
```

Comment:
All previous token are not going to have this.  Should you just omit it and it will become the "default".